docs/spec/spec.md

```diff
@@ -204,34 +204,25 @@ channel._
 | apiVersion | String       | API version of backing channel CRD                  |             |
 | spec       | RawExtension | Spec to be passed to backing channel implementation |             |
 
-### eventing.SubscriberSpec
+### Destination
 
-| Field               | Type            | Description | Constraints              |
-| ------------------- | --------------- | ----------- | ------------------------ |
-| ref<sup>1</sup>     | ObjectReference |             | Must adhere to Callable. |
-| dnsName<sup>1</sup> | String          |             |                          |
-
-1: One of (ref, dnsName), Required.
-
-### duck.SubscriberSpec
-
-| Field         | Type   | Description                                                 | Constraints    |
-| ------------- | ------ | ----------------------------------------------------------- | -------------- |
-| uid           | String | The Subscription UID this SubscriberSpec was resolved from. |                |
-| subscriberURI | String | The URI name of the endpoint for the subscriber.            | Must be a URL. |
-| replyURI      | String | The URI name of the endpoint for the reply.                 | Must be a URL. |
-
-### pkg/duck.Destination
-
-| Field           | Type            | Description                                                                                          | Constraints    |
-| --------------- | --------------- | ---------------------------------------------------------------------------------------------------- | -------------- |
-| ref<sup>1</sup> | ObjectReference | The Subscription UID this SubscriberSpec was resolved from.                                          |                |
-| uri<sup>1</sup> | String          | Either an absolute URL (if ref is not specified). Resolved using the base URI from ref if specified. | Must be a URL. |
+| Field           | Type            | Description                                                                                          | Constraints                   |
+| --------------- | --------------- | ---------------------------------------------------------------------------------------------------- | ----------------------------- |
+| ref<sup>1</sup> | ObjectReference | The Subscription UID this SubscriberSpec was resolved from.                                          | Must adhere to _Addressable_. |
+| uri<sup>1</sup> | apis.URL        | Either an absolute URL (if ref is not specified). Resolved using the base URI from ref if specified. | Must be a URL.                |
 
 1: One or both (ref, uri), Required. If only uri is specified, it must be an
 absolute URL. If both are specified, uri will be resolved using the base URI
 retrieved from ref.
 
+### duck.SubscriberSpec
```

Comment:
You have duck here, but in Destination you don't, I felt it was important to have the duck there to indicate it's not a real type.

Reply:
SubscriberSpec is an interface, but Destination is just a simple struct, It happens to be shared and implemented in pkg but that is an internal detail for knative, and I did not think it is part of the spec. We could copy over Destination into eventing and keep the shape, and still be following the spec.